### PR TITLE
[vnet] leaf cluster support

### PR DESCRIPTION
**Pull request by nklaassen**
This PR adds support for accessing apps in leaf clusters to VNet.

### CODE REVIEW

**Review by ravicious — COMMENTED** [review]
The code looks fine. I'll give it a go on Monday, since I don't have much time left today and I'll need to configure a leaf cluster on my self-hosted cluster.

Though I wonder if I could test VNet on local clusters if I used dnsmasq for them instead of `/etc/hosts`.

It did set up `/etc/resolver` correctly for my weird local leaf cluster where I purposefully set cluster name to be different than it's public addr of the proxy service.

**Review by ravicious — APPROVED** [review]
I tested it with a real setup and it works just fine.

I couldn't use dnsmasq to test it with local clusters, as dnsmasq also requires you to put files in `/etc/resolver` on macOS. 🙃

**Review by smallinsky — APPROVED** [review]

File: lib/vnet/app_resolver.go [hunk 1]
```diff
@@ -155,63 +152,106 @@ func (r *TCPAppResolver) ResolveTCPHandler(ctx context.Context, fqdn string) (*T
...
-	if err != nil {
-		r.slog.ErrorContext(ctx, "Failed to get teleport client, not checking custom DNS zones.", "profile", profileName, "error", err)
-		return false, nil
-	}
-	clusterName := rootClient.ClusterName()
-	for _, zone := range vnetConfig.GetSpec().GetCustomDnsZones() {
-		if !isSubdomain(fqdn, zone.GetSuffix()) {
-			// The queried app fqdn is not a subdomain of this custom zone suffix, skip it.
+	allClusters := append([]string{""}, leafClusters...)
+	for _, leafClusterName := range allClusters {
```
[the remark this answers is not in the document]
Comment (nklaassen) [reply lib/vnet/app_resolver.go:204]: the empty string leaf cluster means it gets a client for the root cluster, I'll add a comment on that

**Comment by nklaassen** [comment]
> I tested it with a real setup and it works just fine.
> 
> I couldn't use dnsmasq to test it with local clusters, as dnsmasq also requires you to put files in `/etc/resolver` on macOS. 🙃

If you want, you can run locally with your proxy addresses in /etc/hosts/, that's what I do, but yeah that wouldn't get coverage if our DNS resolver messes up on proxy addresses. Maybe using different versions of a localhost nip.io public_addr for each cluster proxy would work. You could also probably configure dnsmasq in your network settings instead of /etc/resolver

**Pull request merged**